Mathlib/Order/Directed.lean

```diff
@@ -321,3 +325,57 @@ instance (priority := 100) OrderTop.to_isDirected_le [LE α] [OrderTop α] : IsD
 -- see Note [lower instance priority]
 instance (priority := 100) OrderBot.to_isDirected_ge [LE α] [OrderBot α] : IsDirected α (· ≥ ·) :=
   ⟨fun _ _ => ⟨⊥, bot_le _, bot_le _⟩⟩
+
+namespace DirectedOn
+
+section Pi
+
+variable {ι : Type*} {α : ι → Type*}
+
+variable (r : (i : ι) →  α i →  α i → Prop)
```

Comment:
I think this should be left implicit: the most common case will be product orders where it will be inferred, and otherwise named arguments should work in our favor.